ledger/src/blockstore.rs

```diff
@@ -1051,6 +1051,26 @@ impl Blockstore {
             });
 
             if erasure_config != erasure_meta.config {
+                let coding_start = erasure_meta.first_coding_index;
+                let coding_end = coding_start + erasure_meta.size as u64;
+                let mut conflicting_shred = None;
+                for coding_index in coding_start..coding_end {
+                    if let Ok(Some(shred)) = self.get_coding_shred(slot, coding_index) {
```

Comment:
Thinking about this more, this check will trigger a duplicate proof even if

Two *different* coding shreds from the same FEC set have *different* configs for one of (num_coding, num_data, first_coding_index), but those coding shreds aren't for the same index. 

I think what we actually want to check for is if two coding shreds have the same slot/index and different payloads. For instance, if a leader generated

C1, C2. for one set, and a second C2' with a different coding config, we would want to detect C2, C2' conflict.

Reply:
We can check both. Changing the coding config in the middle of an FEC set should be a violation, right?